### PR TITLE
Add support for `rawTerm` to Python bindings

### PR DESCRIPTION
**Pull request by Baltoli**
~~Blocked on https://github.com/runtimeverification/llvm-backend/pull/907~~

This PR follows up #907 by having the Python bindings strip patterns of the form `rawTerm{}(inj{...}(...)})` when deserializing from binary KORE; this will allow Pyk and subsequently the proof checker's parser to soundly load function arguments from proof traces and pass them to the function evaluator in #905.

### CODE REVIEW

File: test/python/test_raw_term.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        with tempfile.NamedTemporaryFile() as fp:
+            term._serialize_raw(fp.name, "SortInt{}")
+            data = fp.read()
+            
+            with_raw = kllvm.ast.Pattern.deserialize(data, strip_raw_term = False)
+            without_raw = kllvm.ast.Pattern.deserialize(data)
+
+            self.assertEqual(str(token), str(without_raw))
+            self.assertNotEqual(str(token), str(with_raw))
+            self.assertTrue(str(token) in str(with_raw))
```
Comment (tothtamas28) [inline test/python/test_raw_term.py:30]:
```suggestion
        self.assertEqual(str(token), str(without_raw))
        self.assertNotEqual(str(token), str(with_raw))
        self.assertTrue(str(token) in str(with_raw))
```

File: test/python/test_raw_term.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    def int_token(self, v):
+        return kllvm.parser.Parser.from_string(f'\\dv{{SortInt{{}}}}("{v}")').pattern()
+
+    def test_raw_terms(self):
+        token = self.int_token(76)
+        term = kllvm.runtime.Term(token)
+
+        with tempfile.NamedTemporaryFile() as fp:
+            term._serialize_raw(fp.name, "SortInt{}")
```
Comment (tothtamas28) [inline test/python/test_raw_term.py:22]: What's the primary way to serialize, is it `InternalTerm.serialize` or `Pattern.serialize`?

File: test/python/test_raw_term.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            data = fp.read()
+            
+            with_raw = kllvm.ast.Pattern.deserialize(data, strip_raw_term = False)
```
[the remark this answers is not in the document]
Comment (tothtamas28) [reply test/python/test_raw_term.py:25]: And what's the primary way to serialize, is it `InternalTerm.serialize` or `Pattern.serialize`?

[the remark this answers is not in the document]
Comment (Baltoli) [reply test/python/test_raw_term.py:25]: One produces a term (runtime structure) and the other produces a pattern (AST structure); it depends what you want to do with the result of deserialization - if you're pretty-printing or turning it back into a Pyk data structure, you probably want `Pattern`, but if you want to take steps on the result then `InternalTerm`.

You can think of some of the operations on the two types being:
```
deserialize_term : BinaryKORE -> Term
deserialize_pattern : BinaryKORE -> Pattern

construct : Pattern -> Term
term_to_pattern : Term -> Pattern

step : Term -> Term
```
where something like a `CompositePattern` is a `Pattern`, and you can't construct a `Term` directly as a data structure (only by building one from a `Pattern` or from binary / textual KORE).

[the remark this answers is not in the document]
Comment (Baltoli) [reply test/python/test_raw_term.py:25]: In fact, on second inspection I think my preference is to leave it only on `Pattern`. The stripping of the raw term is really an AST manipulation, and the less of that kind of code we have to duplicate and maintain at the `Term` level the better.

**Review by Robertorosmaninho — APPROVED** [review]
Assuming that the changes on the function signatures don't break anything on pyk and on python's bindings, The c++ part LGTM!

**Pull request merged**